scripts/lib/setup-apt-repo-debathena

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+set -x
+
+SOURCES_FILE=/etc/apt/sources.list.d/zulip.list
+STAMP_FILE=/etc/apt/sources.list.d/zulip.list.apt-update-in-progress
+zulip_source_hash=$(sha1sum "$SOURCES_FILE")
+
+apt-get install -y lsb-release apt-transport-https gnupg
+
+SCRIPTS_PATH="$(dirname "$(dirname "$0")")"
+
+release=$(lsb_release -sc)
+if [ "$release" = "xenial" ]; then
+    apt-key add "$SCRIPTS_PATH"/setup/debathena-archive.asc
+    cat >/etc/apt/sources.list.d/zulip.list <<EOF
+    deb http://debathena.mit.edu/apt $release debathena debathena-config
+    deb-src http://debathena.mit.edu/apt $release debathena debathena-config
+EOF
+else
+    echo "Unsupported release $release."
+    exit 1
+fi
+
+# Copied blindly from scripts/lib/setup-apt-repo
+if [ "$zulip_source_hash" = "$(sha1sum "$SOURCES_FILE")" ] && ! [ -e "$STAMP_FILE" ]; then
+    echo "zulip.list file did not change; skipping apt-get update"
+else
+    # We create this stamp file to ensure `apt-get update` will be run
+    # the next time this script is invoked, and each time after, until
+    # `apt-get update` finishes successfully.
+    touch "$STAMP_FILE"
+    apt-get update && rm -f "$STAMP_FILE"
+fi
```

Comment:
So, there's a preparatory project I'd like to do before we copy this file, which is to add `set -e` to `setup-apt-repo`.  

Also, the copying is probably wrong, in that `SOURCES_FILE` and `STAMP_FILE` should presumably be renamed to have `debathena` in their name (or something).

Reply:
(Getting `set -e` working is a bit of work, since we need to add handling for e.g. `$SOURCES_FILE` not existing)

Reply:
Sounds good.